google/cloud/spanner_v1/_opentelemetry_tracing.py

```diff
@@ -18,15 +18,40 @@
 
 from google.api_core.exceptions import GoogleAPICallError
 from google.cloud.spanner_v1 import SpannerClient
+from google.cloud.spanner_v1 import gapic_version
 
 try:
     from opentelemetry import trace
     from opentelemetry.trace.status import Status, StatusCode
+    from opentelemetry.semconv.attributes.otel_attributes import (
+       OTEL_SCOPE_NAME,
+       OTEL_SCOPE_VERSION,
+    )
 
     HAS_OPENTELEMETRY_INSTALLED = True
 except ImportError:
     HAS_OPENTELEMETRY_INSTALLED = False
 
+LIB_VERSION = gapic_version.__version__
+LIB_FQNAME = 'cloud.google.com/python/spanner'
+TRACER_NAME = LIB_FQNAME
+TRACER_VERSION = LIB_VERSION
+
+
+def get_tracer(tracer_provider=None):
+    """
+    get_tracer is a utility to unify and simplify retrieval of the tracer, without
+    leaking implementation details given that retrieving a tracer requires providing
+    the full qualified library name and version.
+    When the tracer_provider is set, it'll retrieve the tracer from it, otherwise
+    it'll fall back to the global tracer provider and use this library's specific semantics.
+    """
+    if not tracer_provider:
+        # Acquire the global tracer provider.
+        tracer_provider = trace.get_tracer_provider()
+
+    return tracer_provider.get_tracer(TRACER_NAME, TRACER_VERSION)
```

Comment:
Why do we need this condition? I think get_tracer() by default returns global tracer if tracer_provider is empty.
Check this code 
![image](https://github.com/user-attachments/assets/1c3ddbb8-c093-43e8-ba42-58b2c4b3678c)

Reply:
I was being explicit, Python isn't particularly known for handling None as a fallback. In your suggestion you are asking me to change that code to
```python
return trace.get_tracer(TRACER_NAME, TRACER_VERSION, tracer_provider)
```
but note that this is much more confusing for readers than
```python
if not tracer_provider:
   tracer_provider = tracer.get_provider()
return tracer_provider.get_tracer(TRACER_NAME, TRACER_VERSION)
```
Should I still change it after my explanation above @harshachinta?

Reply:
@surbhigarg92 
any thoughts here?